packages/appsync-modelgen-plugin/src/preset.ts

```diff
@@ -346,5 +349,31 @@ export const preset: Types.OutputPreset<AppSyncModelCodeGenPresetConfig> = {
           )}`,
         );
     }
-  },
-};
+  };
+
+/**
+ * @internal
+ */
+export const presetSync: SyncTypes.OutputPreset<AppSyncModelCodeGenPresetConfig> = {
+  buildGeneratesSection: (options: SyncTypes.PresetFnArgs<AppSyncModelCodeGenPresetConfig>): SyncTypes.GenerateOptions[] => {
+    const {cache, pluginMap, ...otherOptions} = options;
+
+    return buildGenerations(otherOptions).map((config: GenerateOptions) => ({
+      pluginMap,
+      cache,
+      ...config,
+    }))
```

Comment:
Document why this is okay.

Reply:
Comments added with the goal that the next dev through will have an easier introduction